client/components/jetpack/connection-health/index.tsx

```diff
@@ -17,33 +22,35 @@ import {
 	XMLRPC_ERROR,
 } from './constants';
 import { ErrorNotice } from './error-notice';
-import { useCheckJetpackConnectionHealth } from './use-check-jetpack-connection-health';
 
 interface Props {
 	siteId: number;
 }
 
 export const JetpackConnectionHealthBanner = ( { siteId }: Props ) => {
 	const translate = useTranslate();
+	const dispatch = useDispatch();
 	const siteIsAutomatedTransfer = useSelector(
 		( state ) => !! isSiteAutomatedTransfer( state, siteId )
 	);
 
-	const {
-		data: jetpackConnectionHealth,
-		isLoading: isLoadingJetpackConnectionHealth,
-		isError: isErrorCheckJetpackConnectionHealth,
-	} = useCheckJetpackConnectionHealth( siteId );
-
-	if (
-		isLoadingJetpackConnectionHealth ||
-		isErrorCheckJetpackConnectionHealth ||
-		jetpackConnectionHealth?.is_healthy
-	) {
+	const isJetpackConnectionHealthAPIError = useSelector( ( state ) =>
+		getJetpackConnectionHealthRequestError( state as AppState, siteId )
+	);
+
+	const jetpackConnectionHealth = useSelector( ( state ) =>
+		getJetpackConnectionHealth( state as AppState, siteId )
+	);
+
+	useEffect( () => {
+		dispatch( requestJetpackConnectionHealthStatus( siteId ) );
+	}, [ dispatch, siteId ] );
+
+	if ( isJetpackConnectionHealthAPIError || jetpackConnectionHealth?.is_healthy ) {
```

Comment:
Did you mean to use `! jetpackConnectionHealth?.jetpack_connection_problem` instead of `jetpackConnectionHealth?.is_healthy` here? I thought we're getting rid of `is_healthy`?

Reply:
I was meant to use the error here. Thanks for catching this one!